### PR TITLE
[WFLY-17156] Implement Support for MicroProfile Telemetry

### PR DESCRIPTION
**Pull request by jasondlee**
Add analysis doc

### CODE REVIEW

File: microprofile/WFLY-17156-MicroProfile-Telemetry.adoc [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+=== QE Contacts
+
+=== Testing By
+
+* [x] Engineering
+* [ ] QE
+
+=== Affected Projects or Components
+The OpenTelemetry support in standard WildFly will be modified as need so as to share as much code as possible between the closely-related modules.
```
Comment (kabir) [inline microprofile/WFLY-17156-MicroProfile-Telemetry.adoc:36]: ...as needed...

File: microprofile/WFLY-17156-MicroProfile-Telemetry.adoc [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
```
Comment (kabir) [inline microprofile/WFLY-17156-MicroProfile-Telemetry.adoc:42]: I don't know for sure but I think we normally check the domain box for these?

File: microprofile/WFLY-17156-MicroProfile-Telemetry.adoc [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+* Automatic instrumentation of REST and REST client requests
+* Manual instrumentation via the `@WithSpan` annotation
+* The injection of various OpenTelemetry APIs:
+** `OpenTelemetry`
+** `Tracer`
+** `Span`
+** `Baggage`
+* Remove the MicroProfile OpenTracing extension from the standard configurations (see `Interoperability` below).
+
```
Comment (bstansberry) [inline microprofile/WFLY-17156-MicroProfile-Telemetry.adoc:60]: I think the Requirements section should cover the relationship to the OpenTelemetry subsystem.

Comment (jasondlee) [reply microprofile/WFLY-17156-MicroProfile-Telemetry.adoc:60]: Updated.

File: microprofile/WFLY-17156-MicroProfile-Telemetry.adoc [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+This project will not remove the support for the existing (and deprecated/removed) OpenTracing specification, as that will be handled in a related JIRA.
+
+=== Relationship with OpenTelemetry Support in Standard WildFly
+
+Standard WildFly has supported OpenTelemetry since WildFly 25, so it already has support of much of what is required by MicroProfile Telemetry, with the missing features being `@WithSpan` support,`Baggage` injection, and MicroProfile Config support. To simplify integration and maintenance, MicroProfile Telemetry will build on that support, with the support for most of these missing features being pushed into the standard WildFly implementation. This will be accomplished by replacing the OpenTelemetry implementation in standard WildFly with the SmallRye OpenTelemetry library.
+
+Standard WildFly, however, can not depend on MicroProfile Config being present, so the MicroProfile Telemetry extenion's primary purpose will be to integrate support for this additional specification to complete the compatibility requirements. This will centralize all OpenTelemetry support into one location, and help insure consistent behavior regardless of which configuration is booted.
```
Comment (fabiobrz) [inline microprofile/WFLY-17156-MicroProfile-Telemetry.adoc:80]: Typo: s/extenion/extension/

**Review by fabiobrz — APPROVED** [review]
Hi @jasondlee - Thanks for the work you've done on this proposal, I don't have any comments, since everything is well explained and detailed in it.
I just pointed a minor typo pout, if you don't mind fixing it.
Approving from the QE perspective, great job!

File: microprofile/WFLY-17156-MicroProfile-Telemetry.adoc (hunk 4 above)
Comment (jasondlee) [reply microprofile/WFLY-17156-MicroProfile-Telemetry.adoc]: Fixed. Thanks.

**Comment by jasondlee** [comment]
> Hi @jasondlee - Thanks for the work you've done on this proposal, I don't have any comments, since everything is well explained and detailed in it. I just pointed a minor typo pout, if you don't mind fixing it. Approving from the QE perspective, great job!

Thanks. All the heavy lifting you did on the other PR made this one much easier! :)

**Pull request merged**